src/Container/ContainerLogItem.php

```diff
@@ -9,33 +9,43 @@
 use ReflectionNamedType;
 use ReflectionParameter;
 use ReflectionUnionType;
+use Reflector;
 
 final readonly class ContainerLogItem
 {
     public function __construct(
         public string $id,
-        public ?ReflectionClass $class = null,
-        public ?ReflectionMethod $method = null,
-        public ?ReflectionParameter $parameter = null,
+        public ?Reflector $subject = null
     ) {
     }
 
     public function __toString(): string
     {
-        return match (true) {
-            $this->parameter !== null => $this->parameterAsString(),
-            $this->method !== null => "{$this->method->getDeclaringClass()->getName()}::{$this->method->getName()}()",
-            $this->class !== null => "{$this->class->getName()}",
+        $subjectType = $this->subject ? $this->subject::class : null;
```

Comment:
This makes much more sense as well… what was I thinking?? 😅

Reply:
I'm not sure. 🙃